app/services/slack_notification_service.rb

```diff
@@ -16,9 +20,54 @@ def send_notification
 
   private
 
-  def notify_open_pull_request
+  def notify_pull_request
     pull_request_url = extra_params[:pull_request][:html_url]
     client = Slack::Web::Client.new
-    client.chat_postMessage(channel: '#code-review', text: pull_request_url)
+    client.chat_postMessage(channel: CHANNEL, text: pull_request_url)
+  end
+
+  def filter_unlabeled_action
+    label_name = extra_params[:github_webhook][:label][:name]
+    notify_pull_request if on_hold? label_name
+  end
+
+  def filter_labeled_action
+    label_name = extra_params[:github_webhook][:label][:name]
+    pull_request_url = extra_params[:pull_request][:html_url]
+    if on_hold? label_name
+      matches = find_message pull_request_url
+      delete_message matches
+    end
+  end
+
+  def filter_opened_action
+    labels = extra_params[:pull_request][:labels]
+    notify_pull_request unless labels&.any? { |label| on_hold? label[:name] }
+  end
+
+  def find_message(text)
+    client_find = Slack::Web::Client.new(token: ENV['SLACK_API_TOKEN'])
+    response = client_find.search_messages(channel: CHANNEL, query: "#{text} in:#{CHANNEL}")
+    response.dig(:messages, :matches)
+  end
+
+  def delete_message(matches)
+    client = Slack::Web::Client.new
+    matches.each do |match|
+      timestamp = match[:ts]
+      client.chat_delete(channel: CHANNEL, ts: timestamp) if timestamp
+    end
+  end
+
+  def add_reaction(matches, reaction_name)
+    client = Slack::Web::Client.new
```

Comment:
why don't you need token here? I would create `client` method with `Slack::Web::Client.new(token: ENV['SLACK_API_TOKEN'])`, so you move the dependency to one place